### ISSUE TITLE
[🐞]: Moonlight & Supplementaries keep crashing Server

### ISSUE THREAD

**Issue by Striker183** [MehVahdJukaar/Supplementaries#1290]
### Before Continuing:

- [X] Check you are using the latest version of the mods and its dependencies
- [X] Always include a latest.log if you are crashing
- [X] Remove mod that enhances Minecraft: Optifine, Sodium, others. The issue persists.
- [X] If you are unsure which mod is the culprit.
Disable all of your mods and enable them 1-2 mods each time to isolate the culprit
- [x] Confirm that there is no existing issue with a similar description submitted in the list of issues.

### Version

1.21.1-FABRIC

### Supplementaries Version

supplementaries-fabric_1.21-3.0.28-beta

### Moonlight Lib Version

moonlight fabric_1.21-2.17.12

### Describe Issue

Hello there,
nothing has changed since my last issue 3 months ago. If we play the current versions of Supplementaries & MoonlightLib on the server, it crashes as soon as someone connects. No idea why, but it's definitely due to the 2 mods because with the "old" version (Supplementaries fabric_1.21-3.0.10-alpha; Moonlight fabric_1.21-2.16.0) it works without any problems.

Latest: https://mclo.gs/i6ohgVZ
Crash: https://mclo.gs/5TY7Vku

**Comment by MehVahdJukaar** [comment]
This is likely due to some mod conflict. You can verify by installing just the mod. Obviously log doesn't say anything useful as it's just saying that"some thing took too long". You can increase max server tick time but that won't solve any underlying issue unless it's an instantaneous one. Only way to resolve this is to first find out the mods it's doing this with

**Comment by dandy-is-lion** [comment]
Not exactly the same issue, but I've been struggling to upgrade an existing world from Supplementaries 2.8.17 to latest (3.1.10) on Forge. It refuses to let me load into the world by hanging on the loading screen indefinitely. No obvious errors in the log. There's hardly anything related to Supplementaries in the existing world, aside from Flax and a Sack, so I'm not sure why it's refusing to load. In a test world I'm able to load in just fine. Downgrading to 2.8.17 lets me load into the existing world just fine as well. Was upgrading because maps were causing crashes with ImmediatelyFast, so I've disabled that temporarily. I know >3.0 fixed a lot of issues, but previous attempts at upgrading has lead to world corruption so I've held off on updating Supplementaries for awhile. Is there something I could edit in in the worldsave that might be causing the hang on the latest version?

The log just in-case: https://mclo.gs/RGoq2yq

Apologies if I'm bumping an unrelated thread. I'll create a new issue report if it's a problem

**Comment by MehVahdJukaar** [comment]
Yeah I don't see much in the log. Try without other mods and add till you find cilprit

**Comment by dandy-is-lion** [comment]
Will do, thank you. I'll try in a minimal instance too to see if I can recreate the hang. This is the only mod I've been struggling to upgrade an existing world with and I've been scratching my head figuring out why

**Comment by RookiesTwo** [comment]
Can reproduce in 1.21.1 neoforge.
When installing C2ME and supplementaries together the issue happens.

https://github.com/RelativityMC/C2ME-neoforge/issues/7

I used [All the leaks](https://github.com/pietro-lopes/AllTheLeaks) to checked the threads not released:

![Image](https://github.com/user-attachments/assets/2bf3892e-a8c3-4b59-b967-38d368cc361b)
